examples/sldeditor/core/widgets/Segment.cpp

```diff
@@ -85,19 +88,77 @@ QRectF Segment::boundingRect() const
         segment_item->property(SegmentItem::P_HEIGHT).toDouble());
 }
 
-void Segment::mouseMoveEvent(QGraphicsSceneMouseEvent *event){
-    _dragged = true;
-
+void Segment::mouseMoveEvent(QGraphicsSceneMouseEvent *event)
+{
     if (segment_item->property(SegmentItem::P_HORIZONTAL).toBool())
         segment_item->setProperty(SegmentItem::P_Y_POS, double(y()+ event->pos().y()));
     else
         segment_item->setProperty(SegmentItem::P_X_POS, double(x()+ event->pos().x()));
-        
 
+    moveHandles();
 }
 
 void Segment::addHandles(Handle* left_handle, Handle* right_handle)
 {
     _left_handle = left_handle;
     _right_handle = right_handle;
+    moveHandles();
+}
+
+void Segment::moveHandles() const
+{
+    // Put the handles in place
+    double x_pos;
+    double y_pos;
+    HandleItem* item;
```

Comment:
Please initialize variables, or, better move declaration closer to their usage (i.e. inside "if" and "else" blocks).